### PR TITLE
Test local mode

### PR DESCRIPTION
**Pull request by genevieveluyt**
Introduces Kind to test local mode.

tldr.
* All tests that ran in manifest mode now also run in local mode
* Added support for only auditing / fixing resources in a specific namespace
* Added support for only auditing non-generated resources

Introducing kind made tests very slow as resources had to be created and destroyed between tests (~20 minutes to run the whole suite) so I changed the tests to run in parallel. This meant the resources for each test had to be isolated, which I decided to do by having them be in their own namespace. I used the filename as the namespace name which meant all the fixture names had to comply with the allowed character set for namespace names (ie. underscores had to be converted to dashes). Using different namespaces also meant adding support for auditing resources in a specific namespace (which is supported by the non-package version of kubeaudit but wasn't yet supported by the package). Creating the resources from fixtures in an actual cluster also meant that additional resources were generated (eg. Deployments also create pods). The “generated” components were slower to be created in the cluster and resulted in flaky tests so I added support to exclude generated resources when running an audit and enabled this for all tests (which is consistent with manifest mode).

### CODE REVIEW

File: auditors/all/all_test.go [hunk 1]
```diff
@@ -49,55 +46,37 @@ func TestAuditAll(t *testing.T) {
...
-					for _, auditResult := range result.GetAuditResults() {
-						if !assert.NotEqual(t, kubeaudit.Error, auditResult.Severity) {
-							return
-						}
+	for _, file := range test.GetAllFileNames(t, fixtureDir) {
+		t.Run(file, func(t *testing.T) {
+			_, report := test.FixSetupMultiple(t, fixtureDir, file, allAuditors)
+			for _, result := range report.Results() {
+				for _, auditResult := range result.GetAuditResults() {
+					if !assert.NotEqual(t, kubeaudit.Error, auditResult.Severity) {
```
[the remark this answers is not in the document]
Comment (genevieveluyt) [reply auditors/all/all_test.go:69]: I think what I was actually going for here was `require` so that the output doesn't get drowned, making it hard to debug. Thanks for catching that!

File: internal/test/test.go [hunk 2]
```diff
@@ -82,18 +91,66 @@ func FixSetupMultiple(t *testing.T, fixtureDir, fixture string, auditables []kub
...
+	require.NoError(err)
+
+	var report *kubeaudit.Report
+	switch mode {
+	case MANIFEST_MODE:
+		manifest, err := os.Open(fixture)
+		require.NoError(err)
+		report, err = auditor.AuditManifest(manifest)
+	case LOCAL_MODE:
+		defer func() {
```
[the remark this answers is not in the document]
Comment (genevieveluyt) [reply internal/test/test.go:107]: Oops I don't think it needs to be. For some reason I thought if it was a function call it would get executed right away and not deferred

File: internal/k8s/client.go [hunk 3]
```diff
@@ -34,14 +38,28 @@ func (kc k8sClient) InClusterConfig() (*rest.Config, error) {
...
 func NewKubeClientLocal(configPath string) (*kubernetes.Clientset, error) {
-	if _, err := os.Stat(configPath); err != nil {
-		return nil, ErrNoReadableKubeConfig
-	}
-	config, err := clientcmd.BuildConfigFromFlags("", configPath)
-	if err != nil {
-		return nil, err
+	var kubeconfig *rest.Config
+	if configPath == "" {
+		var err error
```
[the remark this answers is not in the document]
Comment (genevieveluyt) [reply internal/k8s/client.go:43]: Because `kubeconfig` is declared outside the scope. With `:=` I think it would initialize `kubeconfig` to the local scope rather than assigning to the outer one. I'll clean this up a bit
